eli5/xgboost.py

```diff
@@ -48,3 +68,222 @@ def explain_weights_xgboost(xgb,
         estimator=repr(xgb),
         method='feature importances',
     )
+
+
+@explain_prediction.register(XGBClassifier)
+@explain_prediction.register(XGBRegressor)
+@singledispatch
+def explain_prediction_xgboost(
+        xgb, doc,
+        vec=None,
+        top=None,
+        target_names=None,
+        targets=None,
+        feature_names=None,
+        vectorized=False):
+    """ Return an explanation of XGBoost prediction (via scikit-learn wrapper).
+    """
+    vec, feature_names = handle_vec(xgb, doc, vec, vectorized, feature_names)
+    if feature_names.bias_name is None:
+        # XGBoost estimators do not have an intercept, but here we interpret
+        # them as having an intercept
+        feature_names.bias_name = '<BIAS>'
+
+    missing = FormattedFeatureName('Missing features')
+    missing_idx = feature_names.add_feature(missing)
+
+    X = get_X(doc, vec, vectorized=vectorized)
+    if sp.issparse(X):
+        # Work around XGBoost issue:
+        # https://github.com/dmlc/xgboost/issues/1238#issuecomment-243872543
+        X = X.tocsc()
+
+    proba = predict_proba(xgb, X)
+
+    names = xgb.classes_ if isinstance(xgb, XGBClassifier) else ['y']
+    display_names = get_target_display_names(names, target_names, targets)
+
+    scores_weights = prediction_feature_weights(
+        xgb, X, feature_names, missing_idx)
+
+    res = Explanation(
+        estimator=repr(xgb),
+        method='decision paths',
+        targets=[],
```

Comment:
@kmike I added description and caveats in a9c692e, does it make sense? Is there anything else we should add?